website/www/site/content/en/documentation/sdks/python-pipeline-dependencies.md

```diff
@@ -123,3 +136,19 @@ If your pipeline uses non-Python packages (e.g. packages that require installati
         --setup_file /path/to/setup.py
 
 **Note:** Because custom commands execute after the dependencies for your workflow are installed (by `pip`), you should omit the PyPI package dependency from the pipeline's `requirements.txt` file and from the `install_requires` parameter in the `setuptools.setup()` call of your `setup.py` file.
+
+## Pre-building SDK container image
+
+In the pre-building step, we install pipeline dependencies on the container image prior to the job submission. This would speed up the pipeline execution.\
+To use pre-building the dependencies from `requirements.txt` on the container image. Follow the steps below.
+1. Provide the container engine. We support `docker` and `cloud_build`(requires a GCP project with Cloud Build API enabled).
+
+       --prebuild_sdk_container_enginer <execution_environment>
+2. To pass a base image for pre-building dependencies, enable this flag. If not, apache beam's base image would be used.
```

Comment:
Note that this may not work on an arbitrary base image, the base image should follow the same contract to install dependencies in a setup_only mode as apache beam's base image https://github.com/apache/beam/blob/master/sdks/python/container/boot.go#L49

Reply:
1. `prebuild_sdk_container_engine` (not enginer)
2. Good point that the container needs to have the official entry point for this to work. I think in all container-customization mechanisms we suggest, one way or another we recommend, to use Beam's boot entry point. 
3. As a part of making prebuilding not experimental, I think we should remove prebuild_sdk_container_engine and just use --sdk_container_image flag for this purpose. i don't see the need for two different flags.

Reply:
+1 to the point 3

Reply:
@y1chi took care of #3 in https://github.com/apache/beam/pull/17032. Thanks, @y1chi .

Reply:
As @y1chi pointed out, it may not work if the user doesn't follow apache beam's contract. But we do instruct them to follow the contract in some way.

So, I assume we can introduce this section as part of the instruction?

Reply:
I would remove #2 now that we don't need a special flag.

Reply:
I feel like we can keep 2 and update the pipeline option to `--sdk_container_image=...`.